src/main/kotlin/com/github/imflog/schema/registry/parser/AvroSchemaParser.kt

```diff
@@ -12,13 +14,26 @@ class AvroSchemaParser(
 ) : SchemaParser(client, rootDir) {
 
     override val schemaType: SchemaType = SchemaType.AVRO
-    override fun resolveLocalReferences(
+    override fun resolveLocalAndRemoteReferences(
         subject: String,
         schemaContent: String,
-        localReferences: List<LocalReference>
+        localReferences: List<LocalReference>,
+        remoteReferences: List<SchemaReference>
     ): String {
         val parser = Schema.Parser()
+
+        resolveRemoteReferences(parser,remoteReferences)
```

Comment:
I don't think this is the way to go. If we do this, the registered schema will flatten all the references and thus only one big schema will be registered in the registry (no more remotely registered schemas).
I am not sure how we could avoid this appart from forbiding users to reference a remote reference in a local reference file. If this condition is always verified, then we don't have the parse ordering issue and we could parse all the local references, flatten them in our base schema and then let registry handle the remote references.
This is a bit more difficult because at one point we will have to deal with raw files and not use the parser anymore.

WDYT ?

Reply:
The use case of a local reference that needs a remote reference is a use case that we have on our side :/.
As well we have some localReference, that reference other localReference, s
So we might work with raw data in json, but a bit more tricky to identify and copy local reference within the schema.
I though about using the "JSONObject", but it does not seems to provide easy method to navigate within the json

Reply:
In this case we shouldn't use the parser at all for local references, just read files and dump the correct part in a consolidated avsc file.
As avsc files can only contain one record, we would need to do JSON operations to replace the parts where a local type is used by it's actual record definition.
It probably imply some kind of recursive algorithm to find all the types that are local (given than remote and local are not easily distinguished in Avro) and try to do the replace. If not found, this is probably a remote dependency.

Reply:
I'll have a look in this direction

Reply:
I have pushed a new version, working with JSONObject and some reflection to move within the JsonTree.
All seems good from the UT point of view.
Let me know what you think

Reply:
Hello @ImFlog .
Did you had time to have a look on the new proposition :) ?

Reply:
I am in holidays this week and thus I will be able to review this starting next week. Sorry for the delay 🙂